### PR TITLE
Update README.md and adding docker-compose.yaml

### PR DESCRIPTION
**Pull request by marianofsl**
- Adding README.md with basic information.
- Adding docker-compose.yaml with postgres container.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,19 +1,80 @@
...
+This app has an almost complete test suite.
+
+The tests to update and delete a cuboid are incomplete, your task is to improve them.
+
+The other tests are valid and you must not modify them. In other words, you need to add the missing functionalities so that these tests pass.
+
+You should also take note of the linter and prettier. The linter is currently passing and must pass on completion
+of the challenge, without any modifications to the config. 
+
+**Note**: The only tests to be modified are tests to update and delete a cuboid. All other tests must remain unchanged.
```
Comment (marianofsl) [inline README.md:26]: I'm not sure if this line is correct

File: README.md [hunk 2]
```diff
@@ -1,19 +1,80 @@
...
+1. Create a pull request to the master branch on your repository.
+1. Invite @bencarle and @mfpiccolo to your private repository.
+1. Send the link to your pull request to signify the completion of the challenge. 
+
+## Technology
+
+This app uses the following key technologies:
+
+- [Phoenix](https://www.phoenixframework.org/)
+
```
Comment (ehegnes-mgm) [inline README.md:50]: I'd add `ecto`, `mix`, and `docker-compose` here just for the sake of completeness.

File: README.md [hunk 3]
```diff
@@ -1,19 +1,83 @@
...
+3. Add a second remote to your local copy of this repository and push the master branch.
+4. Checkout a feature branch where you will make your changes.
+5. Setup the app and get it running. Verify that the linter passes and the test suite fails.
+6. Implement tests to update and delete a cuboid.
+7. Add missing functionalities so the other tests pass. Do NOT modify these tests.
+8. Commit as appropriate as you complete the challenge. (More than one commit is expected)
+9. Push your committed changes to your repository on your feature branch. 
+10. Create a pull request to the master branch on your repository.
+11. Invite @bencarle and @mfpiccolo to your private repository.
+12. Send the link to your pull request to signify the completion of the challenge.
```
Comment (marianofsl) [inline README.md:43]: Is this correct?

File: README.md [hunk 4]
```diff
@@ -1,19 +1,80 @@
...
+
+This app uses the following key technologies:
+
+- [Phoenix](https://www.phoenixframework.org/)
+
+
+## Setup
+
+Install dependencies.
+```shell script
```
Comment (ehegnes-mgm) [inline README.md:55]: GitHub uses `linguist` to parse code blocks. `shell script` is [not valid](https://github.com/github/linguist/blob/master/lib/linguist/languages.yml), but `shell` is.
```suggestion
```shell
```

File: README.md [hunk 5]
```diff
@@ -1,19 +1,80 @@
...
+mix deps.get
+```
+
+Create and migrate your database.
+```shell script
+mix ecto.setup
+```
+Run ```docker-compose``` up to start the PostgreSQL database, or look inside ```config/dev.exs``` for database configuration requirements.
```
Comment (ehegnes-mgm) [inline README.md]:
```suggestion
Run `docker-compose` up to start the PostgreSQL database, or look inside `config/dev.exs` for database configuration requirements.
```
No need for these to be blocks, just inline code.

File: docker-compose.yaml [hunk 6]
```diff
@@ -0,0 +1,11 @@
...
+services:
+  db:
+    image: postgres:13-alpine
+    restart: always
+    environment:
+      POSTGRES_USER: postgres
+      POSTGRES_PASSWORD: postgres
+      POSTGRES_DB: app_dev
+    ports:
+      - 5432:5432
```
Comment (ehegnes-mgm) [inline docker-compose.yaml:11]: No newline at EOF.

**Review by ehegnes-mgm — COMMENTED** [review]
A few nits.

File: README.md (hunk 3 above)
Comment (ehegnes-mgm) [reply README.md:43]: Awesome 👍

File: README.md (hunk 1 above)
Comment (ehegnes-mgm) [reply README.md:26]: Elixir doesn't have proper built-in static analysis, which is an oversight. Options like [`credo`](https://github.com/rrrene/credo) do exist. It does come with a formatter (`mix format`).

We should run a static analysis tool on this anyway and probably add a line that it must pass in the final solution.

File: README.md [hunk 7]
```diff
@@ -1,19 +1,91 @@
...
-  * Forum: https://elixirforum.com/c/phoenix-forum
-  * Source: https://github.com/phoenixframework/phoenix
+This app has an almost complete test suite.
+
+The tests to update and delete a cuboid are incomplete, your task is to improve them.
+
+The other tests are valid and you must not modify them. In other words, you need to add the missing functionalities so that these tests pass.
+
+You should also take note of the linter and prettier. The linter is currently passing and must pass on completion
+of the challenge, without any modifications to the config.
```
Comment (ehegnes-mgm) [inline README.md:24]:
```suggestion
You should also take note of the formatter (`mix format`) and static analysis tooling (`mix credo`). These are currently
passing and must pass on completion of the challenge without any modifications to the config.
```

File: README.md [hunk 8]
```diff
@@ -1,19 +1,91 @@
...
-  * Forum: https://elixirforum.com/c/phoenix-forum
-  * Source: https://github.com/phoenixframework/phoenix
+This app has an almost complete test suite.
+
+The tests to update and delete a cuboid are incomplete, your task is to improve them.
+
+The other tests are valid and you must not modify them. In other words, you need to add the missing functionalities so that these tests pass.
+
+You should also take note of the linter and prettier. Although Elixir doesn't have proper built-in static analysis,
+tools like `mix credo` and `mix format` should be used to ensure pass on completion of the challenge.
```
Comment (ehegnes-mgm) [inline README.md:24]: "prettier" here is not valid for this Elixir project–it should read "formatter". This language is from the Node repo, which is fine, but that project actually uses the `prettier` package to format code, whereas this uses the built-in formatter from `mix`.

```suggestion
Take note of the formatter (`mix format`) and static analysis tooling (`mix credo`). These tools should pass upon on completion of the challenge without any changes to project configuration.
```

**Pull request merged**